lib/pmdtester/cmd.rb

```diff
@@ -0,0 +1,19 @@
+require 'open3'
+
+module PmdTester
+  class Cmd
+    def self.execute(cmd)
+      puts cmd
+
+      begin
+      stdin, stdout, stderr = Open3.popen3(cmd)
+      result = stdout.gets
+      result.chomp! unless result.nil?
+      [result, stderr.gets]
+      rescue Exception => e
```

Comment:
I'm not sure if we're going to handle exception this way.

Reply:
Were would the exception be raised? By the Open3.popen3 call? I guess, that means, that PMD either wasn't executed at all or not correctly, so we can't trust the result (if there is any).
Just printing the exception is fine for now. We might want to add a more speaking message/warning, such as "PMD execution failed. No results. Reason: " + e.message.

Reply:
> Were would the exception be raised? By the Open3.popen3 call?

Refer to the [link](https://stackoverflow.com/questions/26040249/why-does-open3-popen3-return-wrong-error-when-executable-is-missing),  if we add a semicolon after CMD, then we don't need to think about the exception of Open3...
This has been bothering me for a long time, and now it's settled:)